frame/bounties/src/lib.rs

```diff
@@ -544,17 +641,33 @@ decl_module! {
 				let bounty = maybe_bounty.take().ok_or(Error::<T>::InvalidIndex)?;
 				if let BountyStatus::PendingPayout { curator, beneficiary, unlock_at } = bounty.status {
 					ensure!(system::Module::<T>::block_number() >= unlock_at, Error::<T>::Premature);
+					// Get bounty account id
 					let bounty_account = Self::bounty_account_id(bounty_id);
 					let balance = T::Currency::free_balance(&bounty_account);
 					let fee = bounty.fee.min(balance); // just to be safe
```

Comment:
This is no longer just to be safe, because the curator can put in subbounties the entire bounty balance.

Reply:
There is not much change to parent bounty function `fn claim_bounty()`.

Please correct me if I am wrong in my interpretation.

I hope your question is about **handling of residual or dust balance of parent bounty (after paying all subbounty) ?**

As part of this function, Master curator is paid with agreed curator fee & 
curator deposit gets unreserved.

As part of parent bounty, If there is any residual or dust balance. Assumption is master curator has to assign treasury address as beneficiary in `fn award_bounty()`, so that residual balance are pushed back to treasury.

Reply:
ah sorry to not give enough detail in my comments,

I just meant that the inner comment `// just to be safe` is no longer relevant IMO because, it can happen that the balance is less than fee in case the master curator gave more than `original_balance - fee` to the subcurators and subbeneficiary.

Just rewriting the commentis fine.
```suggestion
					let fee = bounty.fee.min(balance); // in case curator gave more to subbounties
```